quantlib/instruments/payoffs.pyx

```diff
@@ -66,7 +67,7 @@ cdef class PlainVanillaPayoff(Payoff):
 
     def __init__(self, option_type, double strike, from_qlpayoff=False):
 
-        if isinstance(option_type, basestring):
+        if isinstance(option_type, str):     # Changed from basestring for Py2/3 compatibility
```

Comment:
it sounds like the most elegant way to do this would to using six.string_types. I'll take that option.